### PR TITLE
Added master volume slider in audio options

### LINKED ISSUE #2614: Need a volume slider
At least having a master volume for now would be useful, then later on when someone feels like it SFX / ambience etc.

There's already a placeholder area for it on the options menu.

Also there should be a cvar to save it so ears aren't blown out every time the game is started.

### PR DESCRIPTION
**Pull request by ManelNavola**
- There is now a slider for changing the master volume
- The slider goes from 0% to 100% in 1% increments, being 0% setting the volume to 0.0f and 100% setting it to 1.0f

Requires space-wizards/RobustToolbox#1427
Fixes #2614

### CODE REVIEW

File: Content.Client/UserInterface/AudioSource.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Content.Client.UserInterface
+{
+    internal class AudioSource
+    {
+    }
+}
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/AudioSource.cs:6]: Unused?

File: Content.Client/UserInterface/OptionsMenu.Audio.cs [hunk 2]
```diff
@@ -1,20 +1,156 @@
...
+            {
+                _cfg.SetCVar(CVars.AudioMasterVolume, MasterVolumeSlider.Value / 100.0f);
+                _cfg.SaveToFile();
+                UpdateChanges();
+            }
+
+            private void OnResetButtonPressed(BaseButton.ButtonEventArgs args)
+            {
+                MasterVolumeSlider.Value = _cfg.GetCVar(CVars.AudioMasterVolume) * 100.0f;
+                MasterVolumeLabel.Text = string.Format("{0:0}%", MasterVolumeSlider.Value);
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/OptionsMenu.Audio.cs:144]: Localize

File: Content.Client/UserInterface/OptionsMenu.Audio.cs [hunk 3]
```diff
@@ -1,20 +1,156 @@
...
+                ResetButton.OnPressed += OnResetButtonPressed;
+                MasterVolumeSlider.OnValueChanged += OnMasterVolumeSliderChanged;
+
+                AddChild(vBox);
+                UpdateChanges();
+            }
+
+            private void OnMasterVolumeSliderChanged(Range range)
+            {
+                MasterVolumeLabel.Text = string.Format("{0:0}%", MasterVolumeSlider.Value);
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/OptionsMenu.Audio.cs:129]: Localize

File: Content.Client/UserInterface/OptionsMenu.Audio.cs [hunk 4]
```diff
@@ -1,20 +1,156 @@
...
+                        }
+                    }
                 });
+
+                MasterVolumeSlider.Value = _cfg.GetCVar(CVars.AudioMasterVolume) * 100.0f;
+                MasterVolumeLabel.Text = string.Format("{0:0}%", MasterVolumeSlider.Value);
+
+                ApplyButton.OnPressed += OnApplyButtonPressed;
+                ResetButton.OnPressed += OnResetButtonPressed;
+                MasterVolumeSlider.OnValueChanged += OnMasterVolumeSliderChanged;
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/OptionsMenu.Audio.cs:121]: Will need to unsubscribe from these as the control is remade every time the round is restarted.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/UserInterface/AudioSource.cs (hunk 1 above)
Comment (ManelNavola) [reply Content.Client/UserInterface/AudioSource.cs]: No clue what this was doing here

File: Content.Client/UserInterface/OptionsMenu.Audio.cs [hunk 5]
```diff
@@ -1,20 +1,162 @@
...
+                AddChild(vBox);
+                UpdateChanges();
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                ApplyButton.OnPressed -= OnApplyButtonPressed;
+                ResetButton.OnPressed -= OnResetButtonPressed;
+                MasterVolumeSlider.OnValueChanged -= OnMasterVolumeSliderChanged;
+                base.Dispose(disposing);
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/OptionsMenu.Audio.cs:130]: In case you haven't just doublecheck it works with round restarts, otherwise LGTM

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**